gtfu/src/main/java/gtfu/tools/TripListGenerator.java

```diff
@@ -336,13 +346,17 @@ public TripListEntry(TripData data, RouteData rdata, String firstStopID, String
         routeID = data.routeID;
         routeShortName = rdata.routeShortName;
         routeLongName = rdata.routeLongName;
+        shapeID = data.shapeID;
 
+        // name = shapeID;
         if (nameField == "headsign"){
             name = headSign;
         } else if (nameField == "route_short_name") {
             name = routeShortName;
         } else if (nameField == "route_long_name") {
             name = routeLongName;
+        } else if (nameField == "shape_id") {
+            name = shapeID;
```

Comment:
You should *always* use `.equals()` for string comparisons instead of `==`. You can get lucky because of string internalization, but we should never count on that.